### PR TITLE
[PFESC-312] Separated out getInputDirs and added conversion failure check

### PR DESCRIPTION
**Pull request by beyondbill**

### CODE REVIEW

**Comment by beyondbill** [comment]
Unit test is on the way!

**Comment by beyondbill** [comment]
Ready for review!

File: quick.go [hunk 1]
```diff
@@ -2,12 +2,14 @@ package fusion
 
 import (
 	"errors"
+	"fmt"
 	"github.com/moovweb/golog"
```
Comment (gregorygtseng) [inline quick.go]: separate this from standard imports

File: quick.go [hunk 2]
```diff
@@ -238,6 +231,48 @@ func (qb *QuickBundlerInstance) gatherFiles(config map[interface{}]interface{})
...
+	inputDirNew := config[":input_directories"]
+	if inputDirNew == nil {
+		inputDirNew = config["input_directories"]
+	}
+	if inputDirNew != nil {
+		v := reflect.ValueOf(inputDirNew)
+		if v.Kind() != reflect.Slice {
+			err = fmt.Errorf(PLURAL_ARRAY_ERR)
+			return inputDirs, err
+		} else {
```
Comment (gregorygtseng) [inline quick.go]: you don't need this else since you return in the if block

File: input_dir_test.go [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+	} else {
+		if err != nil {
+			t.Fatalf("Expected no error. But received an error: %s", err.Error())
+		} else {
+			resultInputDirs := make([]string, len(inputDirs))
+			for i := 0; i < len(inputDirs); i++ {
+				if str, ok := inputDirs[i].(string); ok {
+					resultInputDirs[i] = str
+				} else {
+					t.Fatalf("Failed to convert at least one element in InputDirs to string!")
```
Comment (gregorygtseng) [inline input_dir_test.go]: Fatal()

**Pull request merged**